### PR TITLE
More Programmer Guide and Library Interface updates for next release

### PR DESCRIPTION
**Pull request by akohlmey**
**Summary**

This contains various updates to the Programmer Guide, the library interfaces, the python module, CMake scripts, and the LAMMPS shell in preparation for the stable release.

**Author(s)**

Axel Kohlmeyer, Temple U

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

The `-DLAMMPS_LIB_NO_MPI` has been replaced with `-DLAMMPS_LIB_MPI` and the meaning is now reversed.

**Implementation Notes**

Individual changes are:
- the LAMMPS shell prints now OS/Compiler information when starting and has a new "mem" command.
- various minor reorganizations and corrections in the documentation of the C-library interface
- add and document missing C-library functions in the Python module
- consolidate two redundant sections documenting the building of the manual.
- various minor tweaks to the documentation
- improve formatting of the description of the reax/c control file 
- Move tools and LAMMPS shell build target code to separate Tools.cmake file
- Add a few more useful outputs to the CMake configuration summary
- Print warning when reading positions from a data file with non-zero image flags to non-periodic dimensions

**Post Submission Checklist**

- [ ] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system
- [ ] The feature has been verified to work with the conventional build system
- [x] The feature has been verified to work with the CMake based build system
- [x] Suitable tests have been added to the unittest tree.
- [x] A package specific README file has been included or updated

### CODE REVIEW

File: doc/src/Build_manual.rst [hunk 1]
```diff
@@ -76,66 +98,79 @@ available:
...
-Once Python 3 is installed, open a Terminal and type
+      *virtualenv*
 
-.. code-block:: bash
+      Once Python 3 is installed, open a Terminal and type
+
+      .. code-block:: bash
 
-   pip3 install virtualenv
+         pip3 install virtualenv
```
Comment (rbberger) [inline doc/src/Build_manual.rst:160]: We should think about updating this to the `venv` module that is part of Python 3. That way we avoid having to install `virtualenv`.

File: unittest/c-library/test_library_open.cpp [hunk 2]
```diff
@@ -1,6 +1,7 @@
 // unit tests creating LAMMPS instances via the library interface
 
 #include "lammps.h"
+#define LAMMPS_LIB_MPI 1
```
Comment (rbberger) [inline unittest/c-library/test_library_open.cpp:4]: Should this be here?

**Review by rbberger — APPROVED** [review]

File: unittest/c-library/test_library_open.cpp (hunk 2 above)
Comment (akohlmey) [reply unittest/c-library/test_library_open.cpp:4]: yes

File: doc/src/Build_manual.rst (hunk 1 above)
Comment (akohlmey) [reply doc/src/Build_manual.rst:160]: won't be able to do this in this PR. too many ripples.

**Review by sjplimp — APPROVED** [review]

**Pull request merged**